app/controllers/application_controller.rb

```diff
@@ -2,4 +2,24 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  def widget_params
+    # read all json schemas
+    jsons = Dir[Rails.root.join('db','json','*.json_schema')].map{ |f| JSON.parse File.read(f) }
+    allowed_keys = []
+
+    # add allowed keys as fields, dictionaries, or hashes depending on their type
+    jsons.map{|j| j['properties'] }.each do |properties|
```

Comment:
I'm not totally clear. I think that it will pass all keys of `baz`, and your spec'ing of string definitely wont matter. I'm going to move this to a class per omar's suggestion and then add some tests for it.